VideoLocker/src/main/java/org/edx/mobile/util/ResourceUtil.java

```diff
@@ -1,64 +1,59 @@
 package org.edx.mobile.util;
 
 import android.content.Context;
+import android.support.annotation.PluralsRes;
+import android.support.annotation.StringRes;
 
 import com.squareup.phrase.Phrase;
 
-import org.edx.mobile.R;
 import org.edx.mobile.base.MainApplication;
 
+import java.util.HashMap;
+import java.util.Set;
+
 
 public class ResourceUtil {
     public static final String QuantityHolder = "quantity";
 
-
-    public static String getResourceString(String name) {
-        Context context = MainApplication.instance().getApplicationContext();
-        return getResourceString(name, context);
-    }
-
-    /**
-     * get the string resources dynamically
-     * @param name
-     * @param context
-     * @return
-     */
-    public static String getResourceString(String name, Context context) {
-        int nameResourceID = context.getResources().getIdentifier(name, "string", context.getApplicationInfo().packageName);
-        if (nameResourceID == 0) {
-            throw new IllegalArgumentException("No resource string found with name " + name);
-        } else {
-            return context.getString(nameResourceID);
-        }
-    }
-
-    public static String getResourceString(int resourceId){
+    public static String getResourceString(@StringRes int resourceId) {
```

Comment:
I think we can get rid of this method, as there is no reason to use the global `Context` to get resources when we have a local one available instead, which might have more specific configurations.